src/main/resources/codegen/java/EntityStateProjectionUnitTest.ftl

```diff
@@ -19,38 +29,209 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 public class ${projectionUnitTestName} {
-
+  private Projection projection;
+  private StateStore store;
+  private Map<String,String> valueToProjectionId;
   private World world;
-  private StateStore stateStore;
-  private MockDispatcher dispatcher;
-  private ${aggregateProtocolName} entity;
 
   @BeforeEach
   public void setUp() {
-    dispatcher = new MockDispatcher();
-    world = World.startWithDefaults("test-state-store-projection");;
-    new StateAdapterProvider(world).registerAdapter(${aggregateProtocolName}State.class, new ${aggregateProtocolName}StateAdapter());
-    stateStore = world.actorFor(StateStore.class, InMemoryStateStoreActor.class, Collections.singletonList(dispatcher));
-    new StatefulTypeRegistry(world).register(new StatefulTypeRegistry.Info(stateStore, ${aggregateProtocolName}State.class, ${aggregateProtocolName}State.class.getSimpleName()));
-    entity = world.actorFor(${aggregateProtocolName}.class, ${entityName}.class, "#1");
+    world = World.startWithDefaults("test-state-store-projection");
+
+    store = world.actorFor(StateStore.class, InMemoryStateStoreActor.class, Arrays.asList(new NoOpDispatcher()));
+
+    projection = world.actorFor(Projection.class, WarbleStateStoreProjection.class, store);
```

Comment:
@hamzajg `WarbleStateStoreProjection` should be replaced by the actual name of the `ProjectionActor` class name of each Aggregate.

Reply:
yes, the Entity State Operation Based Projection fixed in this PR #114